src/aria2p/cli.py

```diff
@@ -481,7 +508,10 @@ def subcommand_add(api: API, uris: List[str] = None, from_file: str = None) -> i
     new_downloads = []
 
     for uri in uris:
-        new_downloads.extend(api.add(uri))
+        if int(position) > 0:
+            new_downloads.extend(api.add(uri, options=options, position=int(position) - 1))
+        else:
+            new_downloads.extend(api.add(uri, options=options, position=int(position)))
```

Comment:
First, let `api.add` handle position being an int or None. Secon, `api.add` could many downloads, so dealing with the position is a bit tricky. I suggest the following:

```suggestion
        created_downloads = api.add(uri, options=options, position=position)
        new_downloads.extend(created_downloads)
        if position is not None:
            position += len(created_downloads)
```